src/bindgen.rs

```diff
@@ -1,55 +1,102 @@
+use std::ffi::OsStr;
 use std::path::{Path, PathBuf};
-use std::process::Command;
 use std::{env, fs};
 
 use anyhow::*;
 
-use crate::pio::project::SconsVariables;
 use crate::utils::OsStrExt;
+use crate::{cargo, cli, cmake, cmd, cmd_output, pio};
 
 pub const VAR_BINDINGS_FILE: &str = "EMBUILD_GENERATED_BINDINGS_FILE";
 
-#[cfg(windows)]
-const EXE_SUFFIX: &str = ".exe";
-
-#[cfg(not(windows))]
-const EXE_SUFFIX: &str = "";
-
-#[cfg(windows)]
-const FS_CASE_INSENSITIVE: bool = true;
-
-#[cfg(not(windows))]
-const FS_CASE_INSENSITIVE: bool = false;
-
 #[derive(Clone, Default, Debug)]
 pub struct Factory {
     pub clang_args: Vec<String>,
     pub linker: Option<PathBuf>,
     pub mcu: Option<String>,
+    pub force_cpp: bool,
+    pub sysroot: Option<PathBuf>,
 }
 
 impl Factory {
-    pub fn from_scons_vars(scons_vars: &SconsVariables) -> Result<Self> {
+    pub fn from_scons_vars(scons_vars: &pio::project::SconsVariables) -> Result<Self> {
+        let clang_args = cli::NativeCommandArgs::new(&scons_vars.incflags)
+            .chain(cli::NativeCommandArgs::new(
+                scons_vars
+                    .clangargs
+                    .as_deref()
+                    .unwrap_or_default(),
+            ))
+            .collect();
```

Comment:
I've not tested this yet. But I guess this would only cause problems if `SconsVariables::incflags` and `clangargs` contained unescaped doube-quotes (`"`) which would be removed by `NativeCommandArgs`.